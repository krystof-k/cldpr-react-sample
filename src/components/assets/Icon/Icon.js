import React, {Component} from 'react';
import PropTypes from 'prop-types';
import icons from 'cloudaper-icons';

import {iconName as getIconName} from '../../../tokens';

import StyledIcon from './Icon.styles';

export default class Icon extends Component {
  static propTypes = {
    /** Sets type of icon */
    name: PropTypes.string.isRequired,

    /** Sets size of icon */
    large: PropTypes.bool,
  };

  render() {
    const {name, large, ...props} = this.props;
    const iconName = getIconName(name) || name;
    const path = icons[iconName];
    const size = large ? '48px' : '24px';

    return (
      <StyledIcon large={large} {...props}>
        <StyledIcon.Svg
          width={size}
          height={size}
          viewBox="0 0 24 24"
          fill="currentcolor"
        >
          <path d={path} />
        </StyledIcon.Svg>
      </StyledIcon>
    );
  }
}
