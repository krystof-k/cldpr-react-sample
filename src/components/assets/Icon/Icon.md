### Examples

Icons:

```jsx
  <div>
    <Icon name="arrow-left" />
    <Icon name="arrow-right" />
    <Icon name="arrow-up" />
    <Icon name="arrow-down" />
    <Icon name="caret-up" />
    <Icon name="caret-down" />
    <Icon name="check-circle-filled" />
    <Icon name="check-circle" />
    <Icon name="check" />
    <Icon name="circle" />
    <Icon name="dot-circle" />
    <Icon name="filter" />
    <Icon name="minus-circle" />
    <Icon name="search" />
    <Icon name="x-circle" />
    <Icon name="x" />
  </div>
```

Large Icons:

```jsx
  <div>
    <Icon name="arrow-left" large />
    <Icon name="arrow-right" large />
    <Icon name="arrow-up" large />
    <Icon name="arrow-down" large />
    <Icon name="caret-up" large />
    <Icon name="caret-down" large />
    <Icon name="check-circle-filled" large />
    <Icon name="check-circle" large />
    <Icon name="check" large />
    <Icon name="circle" large />
    <Icon name="dot-circle" large />
    <Icon name="filter" large />
    <Icon name="minus-circle" large />
    <Icon name="search" large />
    <Icon name="x-circle" large />
    <Icon name="x" large />
  </div>
```

Icon with token:

```jsx
<div>
  <Icon name="pagination.next" />
  <Icon name="pagination.prev" />
</div>
```
