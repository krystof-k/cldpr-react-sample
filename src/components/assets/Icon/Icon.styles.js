import styled from 'styled-components';

const Svg = styled.svg``;

const Icon = styled.span`
  display: inline-block;
  width: ${props => (props.large ? '48px' : '24px')};
  height: ${props => (props.large ? '48px' : '24px')};
  vertical-align: middle;

  & > ${Svg} {
    width: ${props => (props.large ? '48px' : '24px')};
    height: ${props => (props.large ? '48px' : '24px')};
    display: block;
  }
`;

Icon.Svg = Svg;

/** @component */
export default Icon;
