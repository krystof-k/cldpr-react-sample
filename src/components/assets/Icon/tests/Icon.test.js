import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('Icon', () => {
  test('renders', () => {
    const json = renderer.create(<Icon name="flag" />).toJSON();
    expect(json).toMatchSnapshot();
  });

  test('large prop', () => {
    const json = renderer.create(<Icon name="flag" large />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule('width', '48px');
    expect(json).toHaveStyleRule('height', '48px');
  });
});
