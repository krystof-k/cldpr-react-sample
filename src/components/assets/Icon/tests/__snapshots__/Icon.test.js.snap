// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon large prop 1`] = `
.c0 {
  display: inline-block;
  width: 48px;
  height: 48px;
  vertical-align: middle;
}

.c0 > .c1 {
  width: 48px;
  height: 48px;
  display: block;
}

<span
  className="c0"
>
  <svg
    className="c1 "
    fill="currentcolor"
    height="48px"
    width="48px"
  >
    <path />
  </svg>
</span>
`;

exports[`Icon renders 1`] = `
.c0 {
  display: inline-block;
  width: 48px;
  height: 48px;
  vertical-align: middle;
}

.c0 > .c1 {
  width: 48px;
  height: 48px;
  display: block;
}

<span
  className="c0"
>
  <svg
    className="c1 "
    fill="currentcolor"
    height="24px"
    width="24px"
  >
    <path />
  </svg>
</span>
`;
